tracer/contrib/net/httptrace/httptrace.go

```diff
@@ -0,0 +1,122 @@
+package httptrace
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// HttpTracer is used to trace requests in a net/http server.
+type HttpTracer struct {
+	tracer  *tracer.Tracer
+	service string
+}
+
+// NewHttpTracer creates a HttpTracer for the given service and tracer.
+func NewHttpTracer(service string, t *tracer.Tracer) *HttpTracer {
+	t.SetServiceInfo(service, "net/http", ext.AppTypeWeb)
+	return &HttpTracer{
+		tracer:  t,
+		service: service,
+	}
+}
+
+// Handler will return a Handler that will wrap tracing around the
+// given handler.
+func (h *HttpTracer) Handler(handler http.Handler) http.Handler {
+	return h.TraceHandlerFunc(http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+		handler.ServeHTTP(writer, req)
+	}))
+}
+
+// TraceHandlerFunc will return a HandlerFunc that will wrap tracing around the
+// given handler func.
+func (h *HttpTracer) TraceHandlerFunc(handler http.HandlerFunc) http.HandlerFunc {
+
+	return func(writer http.ResponseWriter, req *http.Request) {
+
+		// bail out if tracing isn't enabled.
+		if !h.tracer.Enabled() {
+			handler(writer, req)
+			return
+		}
+
+		// trace the request
+		tracedRequest, span := h.trace(req)
+		defer span.Finish()
+
+		// trace the response
+		tracedWriter := newTracedResponseWriter(span, writer)
+
+		// run the request
+		handler(tracedWriter, tracedRequest)
+	}
+}
+
+// span will create a span for the given request.
+func (h *HttpTracer) trace(req *http.Request) (*http.Request, *tracer.Span) {
+	resource := req.Method + " " + req.URL.Path
```

Comment:
Thanks @furmmon! That is awesome. I think the main issue would be resource pattern matching when a parameter is in the path: `www.api.com/apps/:app_id/get_resource_info` In this case the methods I provide would not be sufficient to pull out the relevant path and instead create a new resource for each call to that endpoint:
```
/apps/1/get_resource_info
/apps/2/get_resource_info
/apps/3/get_resource_info
```

In this case I think it may be best to have path be absolute in this base case (IE the way I described) but also allow individual libraries to supply the path. (For use with gomux & httprouter)